File: intellij-lsp/src/com/github/gtache/lsp/contributors/inspection/LSPInspectionProvider.scala
package com.github.gtache.lsp.contributors.inspection

import com.intellij.codeInspection.{InspectionToolProvider, LocalInspectionTool}

/**
  * The provider for the LSP Inspection
  * Returns a single class, LSPInspection
  */
class LSPInspectionProvider extends InspectionToolProvider {

  override def getInspectionClasses: Array[Class[_ <: LocalInspectionTool]] = {
    Array(classOf[LSPInspection])
  }
}

File: intellij-lsp/src/main/kotlin/com/github/gtache/lsp/contributors/inspection/LSPInspectionProvider.kt
package com.github.gtache.lsp.contributors.inspection

import com.intellij.codeInspection.InspectionToolProvider
import com.intellij.codeInspection.LocalInspectionTool

/**
 * The provider for the LSP Inspection
 * Returns a single class, LSPInspection
 */
class LSPInspectionProvider : InspectionToolProvider {

    override fun getInspectionClasses(): Array<Class<out LocalInspectionTool>> {
        return arrayOf(LSPInspection::class.java)
    }
}